### ISSUE TITLE
Image upload occasionally fails and seems to return HTML

### ISSUE THREAD

**Issue by RyanONeill1970** [cloudinary/CloudinaryDotNet#129]
The following (hopefully simple and obvious) code works, but once in a while will glitch with an error. On retrying, the error has gone.

```
private string UploadImageToCloud(string path, string imageId, ImageUploadStatus imageUpload)
{
	try
	{
		var uploadParams = new ImageUploadParams()
		{
			File = new FileDescription(path),
			Invalidate = imageUpload == ImageUploadStatus.Updated,
			PublicId = imageId,
			Overwrite = true,
		};

		var uploadResult = cloudinary.Upload(uploadParams);
		if (uploadResult.StatusCode == System.Net.HttpStatusCode.OK)
		{
			return uploadResult.Version;
		}
	}
	catch (Exception ex)
	{
		Log.Error(ex, "Unable to upload image.");
	}
	return string.Empty;
}
```

It seems that there is a connectivity issue or the backend throws an error and returns the web server HTML error page which causes the JSON parser to throw an error (I've come across the same problem with other code before).

The stack trace is;

> Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: <. Path '', line 0, position 0.
>    at Newtonsoft.Json.JsonTextReader.ParseValue()
>    at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
>    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
>    at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
>    at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
>    at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
>    at CloudinaryDotNet.Api.Parse[T](Object response)
>    at CloudinaryDotNet.Api.CallAndParse[T](HttpMethod method, String url, SortedDictionary`2 parameters, FileDescription file, Dictionary`2 extraHeaders)
>    at CloudinaryDotNet.Cloudinary.Upload(ImageUploadParams parameters)
>    at ImageSyncService.Logic.ImageSync.UploadImageToCloud(String path, String imageId, ImageUploadStatus imageUpload) in C:\inetpub\wwwroot\ImageSyncService\Logic\ImageSync.cs:line 221

I appreciate that the server should not be returning HTML when erroring (or it might be an intermediate gateway) and that's outside the scope of this client, but could the client be made to be more tolerant of this situation by returning a 'server is talking rubbish' message?

If so, do you want a pull request to handle this?

Regards

Ryan

**Comment by roeeba** [comment]
Hi @RyanONeill1970. It does seem to be an issue. Thanks for letting us know. We'll update here once resolved.

**Comment by roeeba** [comment]
Hi @RyanONeill1970, to help us debug the issue further, can you please share your cloud name and the timeframe in which the error occurred?

**Comment by RyanONeill1970** [comment]
Wednesday, August 29, 2018 3:47:01 PM (UK Time)
name: outdoor-and-country

**Comment by roeeba** [comment]
@RyanONeill1970 The issue is resolved on the latest version (1.6.0).
